### PR TITLE
use typos instead of codespell

### PR DESCRIPTION
**Pull request by harshavardhana**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
use typos instead of codespell

## Motivation and Context
just cleanup

## How to test this PR?
install `typos` tool

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [x] Unit tests added/updated
- [x] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/bucket-handlers_test.go [hunk 1]
```diff
@@ -580,7 +580,7 @@ func testListBucketsHandler(obj ObjectLayer, instanceType, bucketName string, ap
 			expectedRespStatus: http.StatusOK,
 		},
 		// Test case - 2.
-		// Test case with invalid accessKey to produce and validate Signature MisMatch error.
+		// Test case with invalid accessKey to produce and validate Signature MissMatch error.
```
Comment (poornas) [inline cmd/bucket-handlers_test.go]: typos doesn't catch case sensitive mistakes

Comment (harshavardhana) [reply cmd/bucket-handlers_test.go]: like?

Comment (harshavardhana) [reply cmd/bucket-handlers_test.go]: @poornas ?

**Comment by harshavardhana** [comment]
typos captures way more spelling mistakes than codespell so that is the intended reason to move to that.

File: cmd/bucket-handlers_test.go (hunk 1 above)
Comment (donatello) [reply cmd/bucket-handlers_test.go]: s/MisMatch/Mismatch/ should fix this - no need to camelcase this.

Comment (donatello) [reply cmd/bucket-handlers_test.go]: MissMatch is not right here @harshavardhana

**Review by donatello — APPROVED** [review]
one comment https://github.com/minio/minio/pull/19088#discussion_r1498443403 - otherwise LGTM

**Review by jiuker — APPROVED** [review]

**Pull request merged**